spec/models/edited_location_field_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+
+RSpec.describe EditedLocationField do
+  describe '#all' do
+    subject { described_class }
+
+    describe '.all' do
+      let(:current) { location(title: 'Beta') }
+      let(:previous) { location(title: 'Alpha') }
+
+      it 'returns an edit record for each field that has changed' do
+        results = subject.all(current, previous)
+        expect(results.count).to eq(1)
+      end
+
+      context 'returned object is correctly populated can on edit details' do
+        it 'for the field that is changed' do
+          results = subject.all(current, previous)
```

Comment:
not at this stage as you are getting all the differences (edits) between the two locations that are passed in.